### PR TITLE
[WIP] Use MLSL in distributed training

### PR DESCRIPTION
**Pull request by AlekseyMarchuk**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- Add usage of the MLSL library in distributed training

### CODE REVIEW

File: NGRAPH_README.md [hunk 1]
```diff
@@ -129,24 +129,25 @@ Please see the files `3rdparty/ngraph_bridge/ngraph.mk` and `make/config.mk` for
              ```
 
 ## Distributed training
-MPI is required for multi-CPU support. Download Open MPI from [here](https://www.open-mpi.org/).
+Multi-CPU distributed training uses Intel MLSL and Intel MPI which are a part of nGraph distribution.
+Environment variables `PATH` and `LD_LIBRARY_PATH` will be updated with path to MPI binaries and libraries during installation.
 
-Use `USE_NGRAPH_DISTRIBUTED=1` and `NGRAPH_EXTRA_CMAKE_FLAGS=-DNGRAPH_DISTRIBUTED_ENABLE=1`
+Use `USE_NGRAPH_DISTRIBUTED=1` and `NGRAPH_EXTRA_CMAKE_FLAGS=-DUSE_NGRAPH_DISTRIBUTED=1`
```
Comment (sfvaroglu) [inline NGRAPH_README.md]: What's the reason for this change? ngraph cmake uses NGRAPH_DISTRIBUTED_ENABLE.

File: NGRAPH_README.md [hunk 2]
```diff
@@ -129,24 +129,25 @@ Please see the files `3rdparty/ngraph_bridge/ngraph.mk` and `make/config.mk` for
...
+make USE_NGRAPH=1 USE_NGRAPH_DISTRIBUTED=1 NGRAPH_EXTRA_CMAKE_FLAGS=-DUSE_NGRAPH_DISTRIBUTED=1 USE_CUDA=0 DEBUG=0 -j
 ```
 
 Here's an example to run ResNet-50 on two nodes:
 
 ``` sh
 export MXNET_NGRAPH_GLUON=1
 export MXNET_ENGINE_TYPE=NaiveEngine
-mpirun -map-by node -x MXNET_NGRAPH_GLUON -x MXNET_ENGINE_TYPE -hostfile hosts -np 2 python MXNET_ROOT/example/image-classification/train_cifar10.py --network resnet --num-layers 50 --kv-store ngraph
+mpirun -hostfile hosts -np 2 python MXNET_ROOT/example/image-classification/train_cifar10.py --network resnet --num-layers 50 --kv-store ngraph
```
Comment (sfvaroglu) [inline NGRAPH_README.md]: MXNET_NGRAPH_GLUON and MXNET_ENGINE_TYPE removed by accident?

**Comment by sfvaroglu** [comment]
Have you tested with ResNet-50? Did you see any performance improvement?

File: NGRAPH_README.md (hunk 2 above)
Comment (AlekseyMarchuk) [reply NGRAPH_README.md]: Intel MPI forwards all environment variables by default, there is no need to list them explicitly

**Comment by AlekseyMarchuk** [comment]
I have a task to test performance with OpenMPI and MLSL. Since new version of MLSL uses several threads for communication, we need to divide CPU cores between OpenMP threads and MLSL threads to prevent their intersection. To do this we need to specify pinning in KMP_AFFINITY like "granularity=fine,proclist=[0-63],explicit", The rest 4 cores are for MLSL (e.g. KNL has 68 cores). At this moment such pinning introduces significant performance degradation, so I'm working on it.

**Comment by AlekseyMarchuk** [comment]
There is no more need in this PR
